File: BraintreeCore/src/test/java/com/braintreepayments/api/UnionPayConfigurationUnitTest.java
package com.braintreepayments.api;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class UnionPayConfigurationUnitTest {

    @Test
    public void fromJson_parsesFullInput() throws JSONException {
        JSONObject input = new JSONObject()
                .put("enabled", true);

        UnionPayConfiguration sut = UnionPayConfiguration.fromJson(input);
        assertTrue(sut.isEnabled());
    }

    @Test
    public void fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        UnionPayConfiguration sut = UnionPayConfiguration.fromJson(null);
        assertFalse(sut.isEnabled());
    }

    @Test
    public void fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        UnionPayConfiguration sut = UnionPayConfiguration.fromJson(new JSONObject());
        assertFalse(sut.isEnabled());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/UnionPayConfigurationUnitTest.kt
package com.braintreepayments.api

import org.robolectric.RobolectricTestRunner
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class UnionPayConfigurationUnitTest {
    @Test
    @Throws(JSONException::class)
    fun fromJson_parsesFullInput() {
        val input = JSONObject()
            .put("enabled", true)
        val (isEnabled) = UnionPayConfiguration(input)
        assertTrue(isEnabled)
    }

    @Test
    fun fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        val (isEnabled) = UnionPayConfiguration(null)
        assertFalse(isEnabled)
    }

    @Test
    fun fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        val (isEnabled) = UnionPayConfiguration(JSONObject())
        assertFalse(isEnabled)
    }
}
